official/coin_build/man/man3/SbVec2f.3
.TH "SbVec2f" 3 "Sun May 28 2017" "Version 4.0.0a" "Coin" \" -*- nroff -*-
.ad l
.nh
.SH NAME
SbVec2f \- The \fBSbVec2f\fP class is a 2 dimensional vector with floating point coordinates\&.
.PP
This vector class is used by many other classes in Coin\&. It provides storage for a vector in 2 dimensions as well as simple floating point arithmetic operations on this vector\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Inventor/SbVec2f\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBSbVec2f\fP (void)"
.br
.ti -1c
.RI "\fBSbVec2f\fP (const float v[2])"
.br
.ti -1c
.RI "\fBSbVec2f\fP (float x, float y)"
.br
.ti -1c
.RI "\fBSbVec2f\fP (const \fBSbVec2d\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP (const \fBSbVec2b\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP (const \fBSbVec2s\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP (const \fBSbVec2i32\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBsetValue\fP (const float v[2])"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBsetValue\fP (float x, float y)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBsetValue\fP (const \fBSbVec2d\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBsetValue\fP (const \fBSbVec2b\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBsetValue\fP (const \fBSbVec2s\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBsetValue\fP (const \fBSbVec2i32\fP &v)"
.br
.ti -1c
.RI "const float * \fBgetValue\fP (void) const"
.br
.ti -1c
.RI "void \fBgetValue\fP (float &x, float &y) const"
.br
.ti -1c
.RI "float & \fBoperator[]\fP (\fBint\fP i)"
.br
.ti -1c
.RI "const float & \fBoperator[]\fP (\fBint\fP i) const"
.br
.ti -1c
.RI "float \fBdot\fP (const \fBSbVec2f\fP &v) const"
.br
.ti -1c
.RI "SbBool \fBequals\fP (const \fBSbVec2f\fP &v, float tolerance) const"
.br
.ti -1c
.RI "float \fBlength\fP (void) const"
.br
.ti -1c
.RI "float \fBsqrLength\fP (void) const"
.br
.ti -1c
.RI "void \fBnegate\fP (void)"
.br
.ti -1c
.RI "float \fBnormalize\fP (void)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBoperator*=\fP (float d)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBoperator/=\fP (float d)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBoperator+=\fP (const \fBSbVec2f\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP & \fBoperator\-=\fP (const \fBSbVec2f\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP \fBoperator\-\fP (void) const"
.br
.ti -1c
.RI "\fBSbString\fP \fBtoString\fP () const"
.br
.ti -1c
.RI "SbBool \fBfromString\fP (const \fBSbString\fP &str)"
.br
.ti -1c
.RI "void \fBprint\fP (FILE *fp) const"
.br
.in -1c
.SS "Protected Attributes"

.in +1c
.ti -1c
.RI "float \fBvec\fP [2]"
.br
.in -1c
.SS "Related Functions"
(Note that these are not member functions\&.) 
.in +1c
.ti -1c
.RI "\fBSbVec2f\fP \fBoperator*\fP (const \fBSbVec2f\fP &v, float d)"
.br
.ti -1c
.RI "\fBSbVec2f\fP \fBoperator*\fP (float d, const \fBSbVec2f\fP &v)"
.br
.ti -1c
.RI "\fBSbVec2f\fP \fBoperator/\fP (const \fBSbVec2f\fP &v, float d)"
.br
.ti -1c
.RI "\fBSbVec2f\fP \fBoperator+\fP (const \fBSbVec2f\fP &v1, const \fBSbVec2f\fP &v2)"
.br
.ti -1c
.RI "\fBSbVec2f\fP \fBoperator\-\fP (const \fBSbVec2f\fP &v1, const \fBSbVec2f\fP &v2)"
.br
.ti -1c
.RI "\fBint\fP \fBoperator==\fP (const \fBSbVec2f\fP &v1, const \fBSbVec2f\fP &v2)"
.br
.ti -1c
.RI "\fBint\fP \fBoperator!=\fP (const \fBSbVec2f\fP &v1, const \fBSbVec2f\fP &v2)"
.br
.in -1c
.SH "Detailed Description"
.PP 
The \fBSbVec2f\fP class is a 2 dimensional vector with floating point coordinates\&.
.PP
This vector class is used by many other classes in Coin\&. It provides storage for a vector in 2 dimensions as well as simple floating point arithmetic operations on this vector\&. 


.PP
\fBSee also:\fP
.RS 4
\fBSbVec2s\fP, \fBSbVec2d\fP, \fBSbVec3s\fP, \fBSbVec3f\fP, \fBSbVec3d\fP, \fBSbVec4f\fP, \fBSbVec4d\fP\&. 
.RE
.PP

.SH "Constructor & Destructor Documentation"
.PP 
.SS "SbVec2f::SbVec2f (void)\fC [inline]\fP"
The default constructor does nothing\&. The vector coordinates will be uninitialized until you do a \fBsetValue()\fP\&. 
.SS "SbVec2f::SbVec2f (const float v[2])\fC [inline]\fP"
Constructs an \fBSbVec2f\fP instance with initial values from \fIv\fP\&. 
.SS "SbVec2f::SbVec2f (float x, float y)\fC [inline]\fP"
Constructs an \fBSbVec2f\fP instance with the initial vector endpoints from \fIx\fP and \fIy\fP\&. 
.SS "SbVec2f::SbVec2f (const \fBSbVec2d\fP & v)\fC [inline]\fP, \fC [explicit]\fP"
Constructs an instance from an \fBSbVec2d\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "SbVec2f::SbVec2f (const \fBSbVec2b\fP & v)\fC [inline]\fP, \fC [explicit]\fP"
Constructs an instance from an \fBSbVec2d\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "SbVec2f::SbVec2f (const \fBSbVec2s\fP & v)\fC [inline]\fP, \fC [explicit]\fP"
Constructs an instance from an \fBSbVec2d\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "SbVec2f::SbVec2f (const \fBSbVec2i32\fP & v)\fC [inline]\fP, \fC [explicit]\fP"
Constructs an instance from an \fBSbVec2d\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SH "Member Function Documentation"
.PP 
.SS "\fBSbVec2f\fP & SbVec2f::setValue (const float v[2])\fC [inline]\fP"
Set new x and y coordinates for the vector from \fIv\fP\&. Returns reference to self\&.
.PP
\fBSee also:\fP
.RS 4
\fBgetValue()\fP\&. 
.RE
.PP

.SS "\fBSbVec2f\fP & SbVec2f::setValue (float x, float y)\fC [inline]\fP"
Set new x and y coordinates for the vector\&. Returns reference to self\&.
.PP
\fBSee also:\fP
.RS 4
\fBgetValue()\fP\&. 
.RE
.PP

.SS "\fBSbVec2f\fP & SbVec2f::setValue (const \fBSbVec2d\fP & v)"
Sets the value from an \fBSbVec2d\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "\fBSbVec2f\fP & SbVec2f::setValue (const \fBSbVec2b\fP & v)"
Sets the value from an \fBSbVec2b\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "\fBSbVec2f\fP & SbVec2f::setValue (const \fBSbVec2s\fP & v)"
Sets the value from an \fBSbVec2s\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "\fBSbVec2f\fP & SbVec2f::setValue (const \fBSbVec2i32\fP & v)"
Sets the value from an \fBSbVec2i32\fP instance\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "const float * SbVec2f::getValue (void) const\fC [inline]\fP"
Returns a pointer to an array of two floats containing the x and y coordinates of the vector\&.
.PP
\fBSee also:\fP
.RS 4
\fBsetValue()\fP\&. 
.RE
.PP

.SS "void SbVec2f::getValue (float & x, float & y) const\fC [inline]\fP"
Returns the x and y coordinates of the vector\&.
.PP
\fBSee also:\fP
.RS 4
\fBsetValue()\fP\&. 
.RE
.PP

.SS "float & SbVec2f::operator[] (\fBint\fP i)\fC [inline]\fP"
Index operator\&. Returns modifiable x or y coordinate\&.
.PP
\fBSee also:\fP
.RS 4
\fBgetValue()\fP and \fBsetValue()\fP\&. 
.RE
.PP

.SS "const float & SbVec2f::operator[] (\fBint\fP i) const\fC [inline]\fP"
Index operator\&. Returns x or y coordinate\&.
.PP
\fBSee also:\fP
.RS 4
\fBgetValue()\fP\&. 
.RE
.PP

.SS "float SbVec2f::dot (const \fBSbVec2f\fP & v) const\fC [inline]\fP"
Calculates and returns the result of taking the dot product of this vector and \fIv\fP\&. 
.SS "SbBool SbVec2f::equals (const \fBSbVec2f\fP & v, float tolerance) const"
Compares the vector with \fIv\fP and returns \fCTRUE\fP if the distance between the vectors is smaller or equal to the square root of \fItolerance\fP\&. 
.SS "float SbVec2f::length (void) const"
Return length of vector\&. 
.SS "float SbVec2f::sqrLength (void) const\fC [inline]\fP"
Returns the square of the length of the vector\&.
.PP
\fBSince:\fP
.RS 4
Coin 2\&.5 
.RE
.PP

.SS "void SbVec2f::negate (void)\fC [inline]\fP"
Negate the vector (i\&.e\&. point it in the opposite direction)\&. 
.SS "float SbVec2f::normalize (void)"
Normalize the vector to unit length\&. Return value is the original length of the vector before normalization\&. 
.SS "\fBSbVec2f\fP & SbVec2f::operator*= (float d)\fC [inline]\fP"
Multiply components of vector with value \fId\fP\&. Returns reference to self\&. 
.SS "\fBSbVec2f\fP & SbVec2f::operator/= (float d)\fC [inline]\fP"
Divides components of vector with value \fId\fP\&. Returns reference to self\&. 
.SS "\fBSbVec2f\fP & SbVec2f::operator+= (const \fBSbVec2f\fP & v)\fC [inline]\fP"
Adds this vector and vector \fIu\fP\&. Returns reference to self\&. 
.SS "\fBSbVec2f\fP & SbVec2f::operator\-= (const \fBSbVec2f\fP & v)\fC [inline]\fP"
Subtracts vector \fIu\fP from this vector\&. Returns reference to self\&. 
.SS "\fBSbVec2f\fP SbVec2f::operator\- (void) const\fC [inline]\fP"
Non-destructive negation operator\&. Returns a new \fBSbVec2f\fP instance which points in the opposite direction of this vector\&.
.PP
\fBSee also:\fP
.RS 4
\fBnegate()\fP\&. 
.RE
.PP

.SS "\fBSbString\fP SbVec2f::toString () const"
Return a string representation of this object 
.SS "SbBool SbVec2f::fromString (const \fBSbString\fP & str)"
Convert from a string representation, return wether this is a valid conversion 
.SS "void SbVec2f::print (FILE * fp) const"
Dump the state of this object to the \fIfile\fP stream\&. Only works in debug version of library, method does nothing in an optimized compile\&. 
.SH "Friends And Related Function Documentation"
.PP 
.SS "\fBSbVec2f\fP operator* (const \fBSbVec2f\fP & v, float d)\fC [related]\fP"
Returns an \fBSbVec2f\fP instance which is the components of vector \fIv\fP multiplied with \fId\fP\&. 
.SS "\fBSbVec2f\fP operator* (float d, const \fBSbVec2f\fP & v)\fC [related]\fP"
Returns an \fBSbVec2f\fP instance which is the components of vector \fIv\fP multiplied with \fId\fP\&. 
.SS "\fBSbVec2f\fP operator/ (const \fBSbVec2f\fP & v, float d)\fC [related]\fP"
Returns an \fBSbVec2f\fP instance which is the components of vector \fIv\fP divided on the scalar factor \fId\fP\&. 
.SS "\fBSbVec2f\fP operator+ (const \fBSbVec2f\fP & v1, const \fBSbVec2f\fP & v2)\fC [related]\fP"
Returns an \fBSbVec2f\fP instance which is the sum of vectors \fIv1\fP and \fIv2\fP\&. 
.SS "\fBSbVec2f\fP operator\- (const \fBSbVec2f\fP & v1, const \fBSbVec2f\fP & v2)\fC [related]\fP"
Returns an \fBSbVec2f\fP instance which is vector \fIv2\fP subtracted from vector \fIv1\fP\&. 
.SS "\fBint\fP operator== (const \fBSbVec2f\fP & v1, const \fBSbVec2f\fP & v2)\fC [related]\fP"
Returns \fI1\fP if \fIv1\fP and \fIv2\fP are equal, \fI0\fP otherwise\&.
.PP
\fBSee also:\fP
.RS 4
\fBequals()\fP\&. 
.RE
.PP

.SS "\fBint\fP operator!= (const \fBSbVec2f\fP & v1, const \fBSbVec2f\fP & v2)\fC [related]\fP"
Returns \fI1\fP if \fIv1\fP and \fIv2\fP are not equal, \fI0\fP if they are equal\&.
.PP
\fBSee also:\fP
.RS 4
\fBequals()\fP\&. 
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for Coin from the source code\&.


insanity/coin/src/base/SbVec2f.cpp
/**************************************************************************\
 * Copyright (c) Kongsberg Oil & Gas Technologies AS
 * All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are
 * met:
 * 
 * Redistributions of source code must retain the above copyright notice,
 * this list of conditions and the following disclaimer.
 * 
 * Redistributions in binary form must reproduce the above copyright
 * notice, this list of conditions and the following disclaimer in the
 * documentation and/or other materials provided with the distribution.
 * 
 * Neither the name of the copyright holder nor the names of its
 * contributors may be used to endorse or promote products derived from
 * this software without specific prior written permission.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
 * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
 * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
\**************************************************************************/

/*!
  \class SbVec2f SbLinear.h Inventor/SbLinear.h
  \brief The SbVec2f class is a 2 dimensional vector with floating point coordinates.
  \ingroup base

  This vector class is used by many other classes in
  Coin. It provides storage for a vector in 2 dimensions aswell
  as simple floating point arithmetic operations on this vector.

  \sa SbVec2s, SbVec2d, SbVec3s, SbVec3f, SbVec3d, SbVec4f, SbVec4d.
*/

#include <Inventor/SbVec2f.h>

#include <limits>
#include <cassert>

#include <Inventor/SbVec2d.h>
#include <Inventor/SbVec2b.h>
#include <Inventor/SbVec2s.h>
#include <Inventor/SbVec2i32.h>
#include <Inventor/fields/SoSFVec2f.h>

#include "coinString.h"

#if COIN_DEBUG
#include <Inventor/errors/SoDebugError.h>
#endif // COIN_DEBUG

#include "tidbitsp.h" // coin_debug_normalize()

// *************************************************************************

/*!
  \fn SbVec2f::SbVec2f(void)

  The default constructor does nothing. The vector coordinates will be
  uninitialized until you do a setValue().
*/

/*!
  \fn SbVec2f::SbVec2f(const float v[2])

  Constructs an SbVec2f instance with initial values from \a v.
*/

/*!
  \fn SbVec2f::SbVec2f(float x, float y)

  Constructs an SbVec2f instance with the initial vector endpoints from
  \a x and \a y.
*/

/*!
  \fn SbVec2f::SbVec2f(const SbVec2d & v)

  Constructs an instance from an SbVec2d instance.

  \since Coin 2.5
*/

/*!
  \fn SbVec2f::SbVec2f(const SbVec2b & v)

  Constructs an instance from an SbVec2d instance.

  \since Coin 2.5
*/

/*!
  \fn SbVec2f::SbVec2f(const SbVec2s & v)

  Constructs an instance from an SbVec2d instance.

  \since Coin 2.5
*/

/*!
  \fn SbVec2f::SbVec2f(const SbVec2i32 & v)

  Constructs an instance from an SbVec2d instance.

  \since Coin 2.5
*/

/*!
  \fn float SbVec2f::dot(const SbVec2f & v) const

  Calculates and returns the result of taking the dot product of this
  vector and \a v.
*/

/*!
  Compares the vector with \a v and returns \c true if the distance
  between the vectors is smaller or equal to the square root of
  \a tolerance.
*/

bool
SbVec2f::equals(const SbVec2f& v, float tolerance) const
{
#if COIN_DEBUG
  if(!(tolerance >= 0.0f))
    SoDebugError::postWarning("SbVec2f::equals",
                              "Tolerance should be >= 0.0f");
#endif // COIN_DEBUG

  float xdist = this->vec[0] - v[0];
  float ydist = this->vec[1] - v[1];

  if((xdist*xdist + ydist*ydist) <= tolerance) return true;
  return false;
}

/*!
  \fn const float * SbVec2f::getValue(void) const

  Returns a pointer to an array of two floats containing the x and y
  coordinates of the vector.

  \sa setValue().
*/

/*!
  \fn void SbVec2f::getValue(float & x, float & y) const

  Returns the x and y coordinates of the vector.

  \sa setValue().
*/

/*!
  Return length of vector.
*/

float
SbVec2f::length(void) const
{
  return static_cast<float>(sqrt(this->sqrLength()));
}

/*!
  \fn float SbVec2f::sqrLength(void) const

  Returns the square of the length of the vector.

  \since Coin 2.5
*/

/*!
  \fn void SbVec2f::negate(void)

  Negate the vector (i.e. point it in the opposite direction).
*/

/*!
  Normalize the vector to unit length. Return value is the original
  length of the vector before normalization.
*/

float
SbVec2f::normalize(void)
{
  float len = this->length();

  if (len > 0.0f) {
    operator/=(len);
  }
#if COIN_DEBUG
  else if (coin_debug_normalize()) {
    SoDebugError::postWarning("SbVec2f::normalize",
                              "The length of the vector should be > 0.0f "
                              "to be able to normalize.");
  }
#endif // COIN_DEBUG
  return len;
}

/*!
  \fn SbVec2f & SbVec2f::setValue(const float v[2])

  Set new x and y coordinates for the vector from \a v. Returns reference to
  self.

  \sa getValue().
*/

/*!
  \fn SbVec2f & SbVec2f::setValue(float x, float y)

  Set new x and y coordinates for the vector. Returns reference to self.

  \sa getValue().
*/

/*!
  Sets the value from an SbVec2d instance.

  \since Coin 2.5
*/

SbVec2f &
SbVec2f::setValue(const SbVec2d & v)
{
  vec[0] = static_cast<float>(v[0]);
  vec[1] = static_cast<float>(v[1]);
  return *this;
}

/*!
  Sets the value from an SbVec2b instance.

  \since Coin 2.5
*/

SbVec2f &
SbVec2f::setValue(const SbVec2b & v)
{
  vec[0] = static_cast<float>(v[0]);
  vec[1] = static_cast<float>(v[1]);
  return *this;
}

/*!
  Sets the value from an SbVec2s instance.

  \since Coin 2.5
*/

SbVec2f &
SbVec2f::setValue(const SbVec2s & v)
{
  vec[0] = static_cast<float>(v[0]);
  vec[1] = static_cast<float>(v[1]);
  return *this;
}

/*!
  Sets the value from an SbVec2i32 instance.

  \since Coin 2.5
*/

SbVec2f &
SbVec2f::setValue(const SbVec2i32 & v)
{
  vec[0] = static_cast<float>(v[0]);
  vec[1] = static_cast<float>(v[1]);
  return *this;
}

/*!
  \fn float & SbVec2f::operator [] (int i)

  Index operator. Returns modifiable x or y coordinate.

  \sa getValue() and setValue().
*/


/*!
  \fn const float & SbVec2f::operator [] (int i) const

  Index operator. Returns x or y coordinate.

  \sa getValue().
*/

/*!
  \fn SbVec2f & SbVec2f::operator *= (float d)

  Multiply components of vector with value \a d. Returns reference to self.
*/


/*!
  \fn SbVec2f & SbVec2f::operator /= (float d)

  Divides components of vector with value \a d. Returns reference to self.
*/

/*!
  \fn SbVec2f & SbVec2f::operator += (const SbVec2f & v)

  Adds this vector and vector \a u. Returns reference to self.
*/

/*!
  \fn SbVec2f & SbVec2f::operator -= (const SbVec2f & v)

  Subtracts vector \a u from this vector. Returns reference to self.
*/

/*!
  \fn SbVec2f SbVec2f::operator - (void) const

  Non-destructive negation operator. Returns a new SbVec2f instance which
  points in the opposite direction of this vector.

  \sa negate().
*/

/*!
  \fn SbVec2f operator * (const SbVec2f& v, float d)
  \relates SbVec2f

  Returns an SbVec2f instance which is the components of vector \a v
  multiplied with \a d.
*/

/*!
  \fn SbVec2f operator * (float d, const SbVec2f & v)
  \relates SbVec2f

  Returns an SbVec2f instance which is the components of vector \a v
  multiplied with \a d.
*/

/*!
  \fn SbVec2f operator / (const SbVec2f& v, float d)
  \relates SbVec2f

  Returns an SbVec2f instance which is the components of vector \a v
  divided on the scalar factor \a d.
*/

/*!
  \fn SbVec2f operator + (const SbVec2f & v1, const SbVec2f & v2)
  \relates SbVec2f

  Returns an SbVec2f instance which is the sum of vectors \a v1 and \a v2.
*/

/*!
  \fn SbVec2f operator -(const SbVec2f & v1, const SbVec2f & v2)
  \relates SbVec2f

  Returns an SbVec2f instance which is vector \a v2 subtracted from
  vector \a v1.
*/

/*!
  \fn int operator == (const SbVec2f & v1, const SbVec2f & v2)
  \relates SbVec2f

  Returns \a 1 if \a v1 and \a v2 are equal, \a 0 otherwise.

  \sa equals().
 */

/*!
  \fn int operator != (const SbVec2f & v1, const SbVec2f & v2)
  \relates SbVec2f

  Returns \a 1 if \a v1 and \a v2 are not equal, \a 0 if they are equal.

  \sa equals().
*/

/*!
  Return a string representation of this object
*/
SbString
SbVec2f::toString() const
{
  return CoinInternal::ToString(*this);
}

/*!
  Convert from a string representation, return wether this is a valid conversion
*/
bool
SbVec2f::fromString(const SbString & str)
{
  bool conversionOk;
  *this = CoinInternal::FromString<SbVec2f>(str,&conversionOk);
  return conversionOk;
}

/*!
  Dump the state of this object to the \a file stream. Only works in
  debug version of library, method does nothing in an optimized
  compile.
*/
void
SbVec2f::print(FILE * fp) const
{
#if COIN_DEBUG
  fputs(this->toString().getString(),fp);
#endif // COIN_DEBUG
}
